generator/integration_tests/tests/golden_kitchen_sink_rest_stub_factory_test.cc

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/integration_tests/golden/internal/golden_kitchen_sink_rest_stub_factory.h"
+#include "google/cloud/common_options.h"
+#include "google/cloud/testing_util/scoped_log.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <generator/integration_tests/test.pb.h>
+#include <gmock/gmock.h>
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace golden_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+
+using ::testing::HasSubstr;
+using ::testing::IsEmpty;
+
+TEST(GoldenKitchenSinkRestStubFactoryTest, DefaultStubWithoutLogging) {
```

Comment:
This (as you probably know) does not test that the metadata decorator is created.  Do you have plans to add that later (say in some kind of "test the full stack" test)?